diff --git a/src/Utility/MimeMailFormatHelper.php b/src/Utility/MimeMailFormatHelper.php
index 4f24f52..09491e8 100644
--- a/src/Utility/MimeMailFormatHelper.php
+++ b/src/Utility/MimeMailFormatHelper.php
@@ -752,14 +752,19 @@ class MimeMailFormatHelper {
     // Overwrite standard headers.
     if ($from) {
       $default_from = $site_config->get('mail');
-      if (!isset($headers['From']) || $headers['From'] == $default_from) {
+      $overwrite = isset($headers['From'], $headers['Sender'], $headers['Return-Path'])
+        && $headers['From'] === $default_from
+        && $headers['Sender'] === $default_from
+        && $headers['Return-Path'] === $default_from;
+
+      if ($overwrite || !isset($headers['From'])) {
         $headers['From'] = $from;
       }
-      if (!isset($headers['Sender']) || $headers['Sender'] == $default_from) {
+      if ($overwrite || !isset($headers['Sender'])) {
         $headers['Sender'] = $from;
       }
       // This may not work. The MTA may rewrite the Return-Path.
-      if (!isset($headers['Return-Path']) || $headers['Return-Path'] == $default_from) {
+      if ($overwrite || !isset($headers['Return-Path'])) {
         $headers['Return-Path'] = '<' . static::mimeMailAddress($from, TRUE) . '>';
       }
     }
